contracts/deployables/freeze-voting/ERC20FreezeVotingV1.sol

```diff
@@ -65,14 +65,14 @@ contract ERC20FreezeVotingV1 is BaseFreezeVotingV1, Version {
     function castFreezeVote() external override {
         uint256 userVotes;
 
-        if (block.number > freezeProposalCreatedBlock + freezeProposalPeriod) {
+        if (block.timestamp > freezeProposalCreated + freezeProposalPeriod) {
             // create a new freeze proposal and set total votes to msg.sender's vote count
 
-            freezeProposalCreatedBlock = uint32(block.number);
+            freezeProposalCreated = uint48(block.timestamp);
 
             userVotes = votesERC20.getPastVotes(
                 msg.sender,
-                freezeProposalCreatedBlock - 1
+                freezeProposalCreated - 1
```

Comment:
Doesn't `getPastVotes` take a block.number as the second param? would this still work passing in a timestamp?

Reply:
it's dependent on whatever the `clock()` on the ERC20 contract returns: a block number or a block timestamp.

if the clock is using block numbers, then the value into `getPastVotes` needs to be a block number.

if the clock is using timestamps, then the value into `getPastVotes` needs to be a timestamp.

Reply:
Ah okay, yep yep `EIP-6372` got it.